src/blocks/kdeconnect.rs

```diff
@@ -238,23 +291,35 @@ impl Device {
         self.device_proxy.name().await.ok()
     }
 
-    async fn battery(&self) -> (Option<u8>, bool) {
-        (
+    async fn battery(&self) -> (Result<u8>, bool) {
+        tokio::join!(
             self.battery_proxy
                 .charge()
-                .await
-                .ok()
-                .map(|p| p.clamp(0, 100) as u8),
-            self.battery_proxy.is_charging().await.unwrap_or(false),
+                .map_ok(|p| p.clamp(0, 100) as u8)
+                .map_err(|_| Error::new("Could not get charge")),
+            self.battery_proxy
+                .is_charging()
+                .map_ok_or_else(|_| false, |x| x),
         )
     }
 
-    async fn notifications(&self) -> Option<usize> {
+    async fn notifications(&self) -> usize {
         self.notifications_proxy
             .active_notifications()
             .await
             .map(|n| n.len())
-            .ok()
+            .unwrap_or(0)
+    }
+
+    async fn network(&self) -> (Result<String>, i32) {
+        tokio::join!(
+            self.connectivity_proxy
+                .cellular_network_type()
+                .map_err(|_| Error::new("Could not get cellular_network_type")),
+            self.connectivity_proxy
+                .cellular_network_strength()
+                .map_ok_or_else(|_| 0, |x| x),
```

Comment:
Is that equivalent to `.unwrap_or(0)`? Also, wouldn't `-1` be a better default?

Reply:
I think it's the equivalent for that, I copied it from the @bim9262 's suggestion. 

I think we actually will never need to worry about this as the dbus signal will always send those two values in a pair. If the `network_type` is not `Ok(t)` we'll just skip it, and if it's present then `network_strength` will have value. 0 is the minimum hence I used it, The -1 part is an undocumented signal that the kdeconnect plugin seems to send if it can't read network status. 

I guess now that we're adding `1` to the value to get the progression icon it makes sense to make it `-1`.